src/utils/address.cljs

```diff
@@ -77,3 +77,47 @@
   [value]
   (when value
     (str (subs value 0 5) "..." (subs value (- (count value) 3) (count value)))))
+
+(defn eip-155-suffix->eip-3770-prefix
+  [eip-155-suffix]
+  (case eip-155-suffix
+    "0x1"    "eth:"
+    "0xa4b1" "arb1:"
+    "0xa"    "oeth:"
+    nil))
+
+(defn is-metamask-address?
+  [address]
+  (re-find regx-metamask-address address))
+
+(defn eip-3770-address?
+  "Checks if address follows EIP-3770 format which is default for Status"
+  [s]
+  (re-find regx-eip-3770-address s))
+
+(defn supported-address?
+  [s]
+  (boolean (or (eip-3770-address? s)
+               (is-metamask-address? s))))
+
+(defn metamask-address->status-address
+  [metamask-address]
+  (when-let [[_ address metamask-network-suffix] (is-metamask-address? metamask-address)]
+    (when-let [status-network-prefix (eip-155-suffix->eip-3770-prefix metamask-network-suffix)]
+      (str status-network-prefix address))))
```

Comment:
IMO the usage of `is-metamaks-address?` isn't a good practice for future maintenance.

I understand that `is-metamaks-address?` will return the result of `re-find`, which is a vector of the matching regex groups, but since this function ends with `?` it's better to always treat it as if its return value is a boolean. 

What I'd suggest is to create a function that performs the string splitting you are performing with `re-find` and another function to just check if an address is a metamask one or not.

wdyt?